spec/unleash/constraint_spec.rb

```diff
@@ -384,24 +388,21 @@
       expect(constraint.matches_context?(context)).to be false
     end
 
-    it 'rejects constraint construction for invalid value types for operator' do
-      array_constraints = ['STR_CONTAINS', 'STR_ENDS_WITH', 'STR_STARTS_WITH', 'IN', 'NOT_IN']
-
-      array_constraints.each do |operator_name|
-        Unleash::Constraint.new('env', operator_name, [])
-        expect do
-          Unleash::Constraint.new('env', operator_name, '')
-        end.to raise_error
-      end
-
-      string_constraints = ['NUM_EQ', 'NUM_GT', 'NUM_GTE', 'NUM_LT', 'NUM_LTE',
-                            'DATE_AFTER', 'DATE_BEFORE', 'SEMVER_EQ', 'SEMVER_GT', 'SEMVER_LT']
-      string_constraints.each do |operator_name|
-        Unleash::Constraint.new('env', operator_name, '')
-        expect do
-          Unleash::Constraint.new('env', operator_name, [])
-        end.to raise_error
-      end
```

Comment:
you can maybe keep the tests but only check that the log message (with the warning) is called?

That is unexpected a different kind of unexpected behavior that would be nice to still have a test for. WDYT?

Reply:
I'm unsure if we should have code for this at all, as this should be handled by unleash. WDYT?

Reply:
I agree that the unleash server should never send us invalid data, but a user could still provide us invalid data via the bootstrap functionality. Additionally the SDK should attempt to be as robust as possible.

So I would really recommend to keep it.

Reply:
Ok will give it a stab

Reply:
Fixed in latest commit:)